cmd/shortener/server/server.go

```diff
@@ -34,7 +41,45 @@ func (s Server) postURLHandle(res http.ResponseWriter, req *http.Request) {
 	res.WriteHeader(http.StatusCreated)
 	_, err = res.Write([]byte(shortenedURL))
 	if err != nil {
-		fmt.Printf("Can not send response from postURLHandle: %s", err)
+		logger.Log().Error("Can not send response from postURLHandle:", zap.Error(err))
+		http.Error(res, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (s Server) apiShortenHandle(res http.ResponseWriter, req *http.Request) {
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		http.Error(res, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	requestData := &dto.ShortenRequest{}
+	err = json.Unmarshal(body, requestData)
+	if err != nil {
+		http.Error(res, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	shortURL := (*s.storage).Store(requestData.URL)
```

Comment:
SUGGESTION: you don't need to do it.
```suggestion
	shortURL := s.storage.Store(requestData.URL)
```